### PR TITLE
Tracking: Add challenge progress tracking

### PR DESCRIPTION
**Pull request by tombettany**
KanoComputing/peldins#2191
Add a class to handle tracking of challenge progress. Originally from
the Make Light project but this is of use in other applications so add
to the library.

cc @convolu @pazdera

### CODE REVIEW

File: kano_profile/challenge_progress_tracker.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+            self._clean_up_playground_tracking()
+
+    def stop_all(self):
+        if self._tracking_mode == ChallengeProgressTracker.CHALLENGE:
+            self.stop_challenge(self._chal_group, self._chal_no)
+        elif self._tracking_mode == ChallengeProgressTracker.PLAYGROUND:
+            self.stop_playground(self._current_hw_ref)
+
+
+challenge_progress_tracker = ChallengeProgressTracker()
```
Comment (convolu) [inline kano_profile/challenge_progress_tracker.py]: I know this is my fault, but before writing it in stone, are you guys ok with the instance being this verbose @tombettany @pazdera ?

**Comment by convolu** [comment]
Other than a call for comments on the length of the variable name holding the instance, it looks ok to me. If you are content :shipit:

File: kano_profile/challenge_progress_tracker.py (hunk 1 above)
Comment (tombettany) [inline kano_profile/challenge_progress_tracker.py]: I'm okay with this name - what do you think @pazdera

Comment (tombettany) [inline kano_profile/challenge_progress_tracker.py]: @convolu @pazdera We are going with `challenge_progress_tracker` if I don't get any other suggestions by 17:00.

Comment (convolu) [inline kano_profile/challenge_progress_tracker.py]: `ch_pr_tr` :-P

Comment (tombettany) [inline kano_profile/challenge_progress_tracker.py]: @convolu You've been spending too much time in C

**Comment by tombettany** [comment]
Okay then, merging this now.

**Pull request merged**